feat(event-types): add team members as optional guests

husniabad opened the pull request:
Adds a setting to event types to automatically invite selected team members as optional attendees to meetings. This does not check their availability. Closes CAL-5091

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #18947 (GitHub issue number)
- Fixes CAL-5091 (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Visual Demo (For contributors especially)

A visual demonstration is strongly recommended, for both the original and new change **(video / image - any one)**.

#### Video Demo (if applicable):

- Show screen recordings of the issue or feature.
- Demonstrate how to reproduce the issue, the behavior before and after the change.

<div>
    <a href="https://www.loom.com/share/2e0cda12e7c9424e83cd3206bd4970ca">
      <p>Add team member as optional guest - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/2e0cda12e7c9424e83cd3206bd4970ca">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/2e0cda12e7c9424e83cd3206bd4970ca-9bdecd0fc5d56b5d-full-play.gif">
    </a>
  </div>
#### Image Demo (if applicable):

- Add side-by-side screenshots of the original and updated change.
- Highlight any significant change(s).

<img width="1579" height="859" alt="image" src="https://github.com/user-attachments/assets/d8f9f90f-183c-4201-b775-96dccffec266" />
<img width="1529" height="836" alt="image" src="https://github.com/user-attachments/assets/fef03170-1fd5-4b10-b533-571e5a6edd22" />


## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR
 /claim #18947 

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

husniabad committed:
feat(event-types): add team members as optional guests

Adds a setting to event types to automatically invite selected team members as optional attendees to meetings. This does not check their availability. Closes CAL-5091

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=22651) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=22651) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=22651) it.</sub>

Devanshusharma2005 reviewed (changes requested):
typecheck issues are failing. Can  you please address them.

husniabad commented:
> typecheck issues are failing. Can you please address them.

Issue fixed, please check again

kart1ka reviewed (changes requested):
Hey @husniabad, Could you pls resolve the merge conflicts?

husniabad committed:
update branch and fix merge conflicts

husniabad commented:
Hi @kart1ka I have updated the branch, could you please review

kart1ka reviewed (dismissed):
LGTM

kart1ka commented:
Good work @husniabad!

husniabad commented:
> Good work @husniabad!

Thank you @kart1ka for approving this PR, I hope you and the team to merge it soon 🙏

husniabad committed:
resolve merge conflicts and test failure

kart1ka reviewed (changes requested):
Left a few questions.

husniabad commented on apps/web/lib/csp.ts:
```diff
@@ -36,9 +36,28 @@ function getCspPolicy(nonce: string) {
 export function getCspNonce() {
-  const nonce = buildNonce(crypto.getRandomValues(new Uint8Array(22)));
+  const BYTE_LEN = 22;
+  const bytes = new Uint8Array(BYTE_LEN);
 
-  return nonce;
+  // Prefer Web Crypto in both browser and modern Node runtimes
+  const hasWebCrypto =
+    typeof (globalThis as any).crypto !== "undefined" &&
+    typeof (globalThis as any).crypto.getRandomValues === "function";
+
+  if (hasWebCrypto) {
+    (globalThis as any).crypto.getRandomValues(bytes);
+  } else {
+    // Node.js environment (including tests)
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const nodeCrypto = require("node:crypto") as typeof import("node:crypto");
+    if (nodeCrypto.webcrypto?.getRandomValues) {
+      nodeCrypto.webcrypto.getRandomValues(bytes);
+    } else {
+      nodeCrypto.randomFillSync(bytes);
+    }
+  }
+
+  return buildNonce(bytes);
```
This one has nothing to do with the implementation feature; I added it because tests were failing due to another merged PR the time I pushed

husniabad committed:
address coderabbitai suggestions

husniabad commented:
Hi @kart1ka I have removed the changes done by me on the file apps/web/lib/csp.ts since it has been fixed from the source, Also I fixed some bugs including CodeRabbitAi suggestion
please check them out

husniabad commented:
> Good work @husniabad!

Hi @kart1ka and @emrysal this PR is ready, and I resolved merge conflicts again, could you please review it!
